flink-python/src/test/java/org/apache/flink/streaming/api/utils/ByteArrayWrapperSerializerTest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.utils;
+
+import org.apache.flink.api.common.typeutils.SerializerTestBase;
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+
+import java.util.Random;
+
+/** Test class for {@link ByteArrayWrapperSerializer}. */
+public class ByteArrayWrapperSerializerTest extends SerializerTestBase<ByteArrayWrapper> {
+
+    private final Random rnd = new Random(346283764872L);
+
+    private static final ByteArrayWrapper EMPTY_ARRAY = new ByteArrayWrapper(new byte[] {});
+
+    @Override
+    protected TypeSerializer<ByteArrayWrapper> createSerializer() {
+        return new ByteArrayWrapperSerializer();
+    }
+
+    @Override
+    protected Class<ByteArrayWrapper> getTypeClass() {
+        return ByteArrayWrapper.class;
+    }
+
+    @Override
+    protected int getLength() {
+        return -1;
+    }
+
+    @Override
+    protected ByteArrayWrapper[] getTestData() {
+        return new ByteArrayWrapper[] {
+            randomByteArray(),
```

Comment:
I think it is better to have the critical cases as dedicated data. The test should include at least one empty array, one short length, one long length (> 256 bytes) and one very long (> 64k). Random arrays can be added, but the important ones should always be in there.

Reply:
It makes sense to add dedicated data test. Regarding to 64k length, it fails with OutOfMemoryError 1 times every two runs in my desktop and so I change it to 32k (succeed running the test case 10 times) for test stability.